.github/workflows/versioning-reset.yml

```diff
@@ -0,0 +1,27 @@
+name: Version reset to dev
+# reset the version if merging a release branch into main
+
+on:
+  push:
+    branches:
+      - main
+      - master
+    paths:
+      - 'app/__version__.py'
```

Comment:
> In principle, this should no longer trigger after this patch is merged with master, right? Since the new release workflow resets __version__ to dev after a tagged commit with versioning is created, and then the __version__.py file is reset which should be identical to what lives in the master branch?

correct, this should not be triggered if merging the head of `stable-vN` onto `master`.
But, this could be triggered for example in the following scenario:
- working on `stable-v2`.
- make release `v2.15` (head of `stable-v2` has `__version__ = "dev"`).
- add some unreleased patches to `stable-v2`.
- decide to merge released v2 patches to `master` (and only those): simplest way is to create a PR from tag `v2.15` into `master`, review it quickly and merge. In this case the `versioning-reset` action will apply immediately after the merge.